src/android/ManagerService.java

```diff
@@ -149,14 +156,26 @@ public void createAndSendEvent(JSONObject obj) {
 
     public void stopServiceIfInactive() {
         if (PendingUpload.count(PendingUpload.class) == 0 && this.connectedPlugin == null) {
-            Intent intent = new Intent(ManagerService.this, ManagerService.class);
```

Comment:
Je ne suis toujours pas d'accord avec ça... Est-ce que on ne veux pas ça ici ?

```
if (this.connectedPlugin == null) {
  if (PendingUpload.count(PendingUpload.class) == 0) {
            Intent intent = new Intent(this, ManagerService.class);
            this.requestObserver.unregister();
            this.requestObserver = null;

            stopService(intent);
   } else {
     if (!isNetworkAvailable) {
       updateNotificationContent("Waiting for connection");
     }
  }
}
```

Si on veut mettre le waiting for connection, tout le temps c'est sur le network observable qu'il faut modifier la notification !

Reply:
@ombr comme #100 ?

Reply:
j'ai déjà essayé ça hier mais le network observable n'est pas dépendant du broadcast donc il met à jour la notification qu'à un court instant ce qui est logique vue que le broadcast continu à tourner. A mon avis ceci ne va pas passer 

```
if (this.connectedPlugin == null) {
  ...
}
```

car si le plugin est toujours rattaché mais sans connexion n'est pas dispo donc y aurai pas de update de notif

Reply:
```
  //  upload restant 0
  //  plugin non rattaché
  if (PendingUpload.count(PendingUpload.class) == 0 && this.connectedPlugin == null) {
            Intent intent = new Intent(this, ManagerService.class);
            this.requestObserver.unregister();
            this.requestObserver = null;

            stopService(intent);
   } else {
   //  upload restant !0
   //  plugin rattaché
   //  connexion non disponible  
     if (!isNetworkAvailable) {
       updateNotificationContent("Waiting for connection");
     }
  }
```

Reply:
j'ai déjà essayé ça hier mais le network observable n'est pas dépendant du broadcast donc il met à jour la notification qu'à un court instant ce qui est logique vue que le broadcast continu à tourner et c'est ce que je suis en train de voir. A mon avis ceci ne va pas passer 

```
if (this.connectedPlugin == null) {
  ...
}
```

car si le plugin est toujours rattaché mais sans connexion n'est pas dispo donc y aurai pas de update de notif

Reply:
j'ai déjà essayé ça hier mais le network observable n'est pas dépendant du broadcast donc il met à jour la notification qu'à un court instant ce qui est logique vue que le broadcast continu à tourner et c'est ce que je suis en train de voir. A mon avis ceci ne va pas passer 

```
if (this.connectedPlugin == null) {
  ...
}
```

car si le plugin est toujours rattaché mais sans connexion donc y aurai pas de update de notif

Reply:
Le code actuel se comporte comme ça si j'ai rien raté:
![image](https://user-images.githubusercontent.com/17636464/82990997-10333880-a00e-11ea-833f-17122b928f95.png)

Est-ce qu'on ne fait pas une méthode comme #101 ?

Reply:
oui car la lib ne persiste pas les tâches restantes donc inProgress dans NotificationHandler se remet à 0 quand on perd la connexion, donc soit on gère via la NotificationHandler mais que je ne conseille pas car faudra mettre isNetworkingAvailable en statique. Soit on adopte la #101 que je trouve logique mais qui n'est pas adapté à @ombr non?

Reply:
C'étais juste pour que la méthode soit complète... on peut l'omettre c'est pas grave.

Reply:
mais on a déjà notificationTitle

Reply:
J'ai remis le title au cas où on aurait besoin de le modifier pour l'attente

Reply:
Il faut que l’on parle de ça aujourd’hui je vous appelle un peu plus tard